alica_standard_library/include/PopulateBlackboard.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include <boost/dll/alias.hpp>
+
+#include <engine/BasicPlan.h>
+#include <engine/BasicUtilityFunction.h>
+
+#include <array>
+#include <string>
+#include <tuple>
+
+namespace alica_standard_library
+{
+
+class PopulateBlackboard : public alica::BasicPlan
+{
+public:
+    PopulateBlackboard(alica::PlanContext& context);
+    static std::unique_ptr<PopulateBlackboard> create(alica::PlanContext& context);
+
+protected:
+    void onInit() override;
+
+private:
+    using KnownTypes = std::tuple<bool, int64_t, uint64_t, double, std::string>;
+    static constexpr std::array<const char*, 5> _knownTypes{"bool", "int64", "uint64", "double", "std::string"};
+
+    void set(const std::string& key, const std::string& type);
+
+    template <std::size_t... Is>
+    void setHelper(const std::string& key, std::size_t typeIndex, std::index_sequence<Is...>)
+    {
+        struct Dummy
```

Comment:
Can you comment to explain what this is doing?

Reply:
this is just syntactic sugar to call `setIfIndex` for every index in the index_sequence.
The comma operator makes sure `setIfIndex` is called, however the value of the expression `setIfIndex, 0) itself is `0`, so basically `dummyFunc` will be called with a bunch of 0s